### DISCUSSION TITLE
"/reload" breaks Editor mode if an Editor Extension doesn't have a shutdown function

### DISCUSSION THREAD

**Discussion by xKingDark** [Mojang/minecraft-editor#58] [Bugs]
**How to Reproduce**:
  **1.** Download this [Example pack](https://github.com/Mojang/minecraft-editor/files/10942855/TestExtension.zip) and apply it to a Project
  **2.** Switch to Crosshair Mode
  **3.** Open chat and execute the following command `/reload`
  **4.** Switch back to Editor Mode

**Observed Result**:
https://user-images.githubusercontent.com/47607185/224095019-838d5679-377c-4bfc-915e-13c77248cbc5.mp4

**Expected Result**:
Editor mode doesn't break even if an extension doesn't have a shutdown function

**Helpful Info**:
- Preview build version: `1.19.80.20`

**Comment by AXOMEstudios** [comment]
Couldn't reproduce the bug. Maybe this happens due to customizations like modified tools, functions etc. on your side?

**Reply by xKingDark** [reply]
I've done some more testing and it looks like /reload breaks editor mode if you have an extension registered

**Comment by minecraftchloe** [comment]
Great find, and I'm going to add this as a test case for us before v0.4 is released. 

You're also using the Editor API before it's ready for prime time 🙃 So, I'm going to update the label as a check when we release v0.4!

**Comment by dacowan** [comment]
Was it just a behavior pack, or did you have a custom resource pack too?

**Reply by xKingDark** [reply]
Just a behavior pack

**Comment by minecraftchloe** [comment]
I _believe_ this has been fixed -- we have `/reload` functioning to refresh scripts as you're building your extension as of #130

**Reply by xKingDark** [reply]
We can't test if it works or not because now `/reload` breaks Editor mode entirely even without extensions but comparing the module to previous versions, nothing changed in the API.

**Reply by conmaster2112** [reply]
Ya, i can confirm current behavior of 1.20.20.20 is that the "/reload" is breaking all Editor Scripts and it applies to Editor Projects vvith Vanilla Only Extension
